x-pack/plugin/security/src/test/java/org/elasticsearch/test/SecurityTestsUtils.java

```diff
@@ -65,6 +83,17 @@ public static void assertThrowsAuthorizationException(LuceneTestCase.ThrowingRun
         assertAuthorizationException(securityException, messageMatcher);
     }
 
+    public static Authentication createApiKeyAuthentication(ApiKeyService apiKeyService, Authentication authentication,
```

Comment:
Something doesn't feel quite right about this.

Let's chat, but I'd prefer a public static method on the `ApiKeyService` (like `createKeyForTesting`) rather than making existing methods public.

Reply:
Yeah, I've ignorant to break the `ApiKeyService` abstraction for test purposes.

I've created a public static inner class `ApiKeyServiceTests.Utils` that translates an authentication for a user to an authentication with an API Key.